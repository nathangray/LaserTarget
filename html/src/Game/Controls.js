/**
 * Component for rendering game controls
 */
import React from 'react';
import * as GAME from './constants.js';
var FontAwesome = require('react-fontawesome');

class Controls extends React.Component {
	propTypes: {
      state:      React.PropTypes.number,
      onChange:   React.PropTypes.func
  }

	constructor(props) {
    super(props);

    // This binding is necessary to make `this` work in the callback
    this.changeHandler = this.changeHandler.bind(this);
  }

	render() {
		switch (this.props.state)	{
			case GAME.State.IDLE:
				return <div className="controls">
						<FontAwesome name='play' data-state={GAME.State.PLAY} onClick={this.changeHandler}/>
					</div>;
			case GAME.State.STARTING:
			case GAME.State.PLAY:
				return <div className="controls">
						<FontAwesome name='stop' data-state={GAME.State.END}  onClick={this.changeHandler}/>
						<FontAwesome name='repeat' data-state={GAME.State.IDLE} onClick={this.changeHandler}/>
					</div>;
			case GAME.State.END:
				return <div className="controls">
					<FontAwesome name='cogs' data-state={GAME.State.IDLE} onClick={this.changeHandler} />
					<FontAwesome name='repeat' data-state={GAME.State.PLAY} onClick={this.changeHandler}/>
				</div>;
			default:
				return <div className="controls">
					<FontAwesome name='add' data-state={GAME.State.IDLE} onClick={this.changeHandler} />
				</div>;
		}
	}

	changeHandler(e) {
		if (typeof this.props.onChange === 'function') {
        this.props.onChange(e.target.dataset.state);
    }
	}
}

export default Controls;
