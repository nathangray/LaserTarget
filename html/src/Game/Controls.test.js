/*
Automated test for Controls component
*/
import React from 'react';
import Controls from './Controls.js';

it("Idle renders correctly", function() {
  const tree =renderer.create(
	   <Controls />
	).toJSON();

	expect(tree).toMatchSnapshot();
});
