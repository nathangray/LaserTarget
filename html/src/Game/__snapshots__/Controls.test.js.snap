// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Idle renders correctly 1`] = `
<div
  className="controls"
>
  <span
    aria-hidden={true}
    className="fa fa-add"
    data-state={0}
    onClick={[Function]}
  />
</div>
`;
